dist/dialog.linker.build.js

```diff
@@ -102,13 +113,24 @@ window.jQuery = window.jQuery || window.shoestring;
 			return;
 		}
 
-		// if someone is calling close directly and the hash for this dialog is in
-		// the url then we need to go back, this will trigger the hashchange binding
-		// in init
-		// NOTE the bindings seem better in the constructor e.g.
-		// "#foo".indexOf("foo") === 1
-		if(window.location.hash.replace(/^#/, "") === this.hash){
-			window.history.back();
+		// if close() is called directly and the hash for this dialog is at the end of the url,
+		// then we need to change the hash to remove it, either by going back if we can, or by adding a history
+		// state that doesn't have it at the end
+		if( window.location.hash.split( "#" ).pop() === this.hash ){
+			// let's check if the first segment in the hash is the same as the first segment in the initial hash
+			// if not, it's safe to use back() to close this out and clean the hash up
+			var firstHashSegment = window.location.hash.split( "#" )[ 1 ];
+			var firstInitialHashSegment = this.initialLocationHash.split( "#" )[ 1 ];
+			if( firstHashSegment && firstInitialHashSegment && firstInitialHashSegment !== firstHashSegment ){
+				window.history.back();
+			}
+			// otherwise, if it's the same starting hash as it was at init time,
+			// we can't trigger back to close the dialog, as it might take us elsewhere.
+			// so we have to go forward and create a new hash that does not have this dialog's hash at the end
+			else {
+				var escapedRegexpHash = this.hash.replace(/([.*+?^=!:${}()|\[\]\/\\])/g, "\\$1");
+				window.location.hash = window.location.hash.replace( new RegExp( "#" + escapedRegexpHash + "$" ), "" );
```

Comment:
I'm confused. The comment says "we can't trigger back to close the dialog" because it might take us away from the page if the page was opened with a dialog hash. So it seems like the idea here is to create a new hash to add a history entry so that if we close the dialog and back from that hash we'll just go back to the page as it was loaded. 

I'm proposing that we handle this in the `close` method. If `close` is called for a dialog that was opened by the initial hash then it should clear that hash which will add an entry to history. This operates under the assumption that we would open the dialog on going back from there if necessary which may not be true/correct.